src/io/memory.cr

```diff
@@ -223,6 +224,19 @@ class IO::Memory < IO
     end
   end
 
+  # :inherit:
+  def getb_to_end : Bytes
+    pos = Math.min(@pos, @bytesize)
+
+    if pos == @bytesize
+      Bytes.new(0)
+    else
+      Slice.new(@buffer + @pos, @bytesize - @pos).dup.tap do
+        @pos = @bytesize
+      end
```

Comment:
I realize this is just copied and adapted from `#gets_to_end`. But I find it much better readable without `tap`:

```suggestion
      @pos = @bytesize
      Slice.new(@buffer + pos, @bytesize - pos).dup
```